app/mailers/producer_mailer.rb

```diff
@@ -79,14 +79,19 @@ def tax_total_from_line_items(line_items)
   def set_customer_data(line_items)
     return unless @coordinator.show_customer_names_to_suppliers?
 
+    @display_business_name = false
     line_items.map do |line_item|
+      customer_code = line_item.order.customer&.code
+      @display_business_name = true if customer_code.present?
```

Comment:
We only want to show the business name column if at least one customer has the code present.